### PR TITLE
docs(tasklist): Tasklist API limitations on Self Managed

### LINKED ISSUE #621: Document Tasklist API limitations on-prem
We need to document that currently tasklist API has some limitations on-prem version

### PR DESCRIPTION
**Pull request by santanaluiz**
Adding Tasklist API limitations to Self Managed environments.

This is a current behavior and can already be merged as soon as reviewed and approved.

Closes #621

### CODE REVIEW

**Comment by felix-mueller** [comment]
lgtm

**Review by felix-mueller — APPROVED** [review]

**Review by ralfpuchert — DISMISSED** [review]

File: docs/self-managed/tasklist-deployment/taskslist-api.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- Clients must create a login request: `POST /api/login?username=<USERNAME>&password=<PASSWORD>`.
+- Use response headers/cookies in the API GraphQL.
+
+### Limitations
+
+As in current Self-Managed versions, we can't differentiate a regular `User` request versus an `API User` request; this imposes some limitations to the Self-Managed Tasklist API:
+
+- Tasks must be claimed by `API User` before any changes are made.
+- Tasks must be assigned to `API User` itself. You cannot assign these tasks to someone else.
+- Tasks can **only** be completed if previously claimed/assigned to `API User`.
```
Comment (christinaausley) [inline docs/self-managed/tasklist-deployment/taskslist-api.md:21]: Is this the same note as the first one claiming "Tasks must be claimed by `API User` before any changes are made."?

**Review by christinaausley — CHANGES_REQUESTED** [review]
Made a few technical adjustments and a comment, including the following:

- Self-Managed instead of Self Managed.
- Utilizing tabs instead of numbers if listing out anything other than a series of steps.

Nice work here. Let me know if you have any questions.

File: docs/self-managed/tasklist-deployment/taskslist-api.md (hunk 1 above)
Comment (akeller) [reply docs/self-managed/tasklist-deployment/taskslist-api.md:21]: @santanaluiz Can you clarify this so we can merge? 🙂

Comment (santanaluiz) [reply docs/self-managed/tasklist-deployment/taskslist-api.md:21]: changes can be adding a variable for example. Completed is another action.

Sorry for the delay on this. I was on vacation :)

**Comment by akeller** [comment]
@christinaausley is this good to merge? Looks like you requested changes but I'm wondering if you did them in your last two commits? 😄

**Comment by christinaausley** [comment]
@akeller From my end, yes. LGTM.

**Comment by santanaluiz** [comment]
sorry, missed this. Let me know if there is any other change I should do myself

Thanks

**Pull request merged**